File: Holiday/kata.py

def is_even(number):
    if number % 2 == 0:
        return True
    elif number % 2 != 0:
        return False

print(is_even(3))

def is_prime(number):
    for count in range(2,number):
        if number % count == 0:
            return False
        else:
            return True
print(is_prime(7))

def substract(number_one, number_two):
    subtraction = number_one - number_two
    return abs(subtraction) 

print(substract(2,4))


def divide(number_one, number_two):
    if number_two == 0:
        return number_two
    else:
        divide = number_one // number_two
        return divide

print(divide(16,4))

def factor_of(number):
    counter = 0
    for count in range(1,number):
        if count % 2 == 0:
            counter +=1
    return counter

print(factor_of(10))

def palindrome(number):
    first_number = number // 10000
    second_number = (number // 1000) % 10
    third_number = (number // 100) % 10
    fourth_number = (number // 10) % 10
    last_number = number % 10
    
    if first_number == last_number and second_number == fourth_number:
        return True
    return False

print(palindrome(54145))

def factorial_of(number):
    factorial = 1
    for count in range(1,number + 1):
        factorial *=count
    return factorial

print(factorial_of(5))


def square_of(number):
    square = number * number
    return square

print(square_of(5))

def is_square(number):
    for count in range(1,number):
        if count * count == number:           
            return True
    return False

print(is_square(25))

File: Holiday/Kata.java
public class Kata{
public static void main(String[] args){

System.out.println(isEven(2));
System.out.println(isPrime(11));
System.out.println(subtract(2,7));
System.out.println(divide(14,7));
System.out.println(factorOf(10));
System.out.println(palindrome(54145));
System.out.println(squareNumber(5));
System.out.println(factorialOf(5));
System.out.println(isSquare(16));
}

public static boolean isEven(int number){
if(number % 2 == 0)
return true;
else if(number % 2 != 0){
}
return false;
}


public static boolean isPrime(int number){

for(int count = 2;count < number;count++)
if(number % count == 0){
return false;
}


return true;
}


public static int subtract(int numberOne,int numberTwo){
int subtraction = numberOne - numberTwo;

return Math.abs(subtraction);
}


public static int divide(int numberOne, int numberTwo){
if(numberTwo == 0){

return numberTwo;
}

int division = numberOne / numberTwo;

return division;
}

public static int factorOf(int number){
int count = 0;
for(int counter =1; counter <= number; counter++){
if(number % counter == 0)
count += 1;
}
return count;
}

public static boolean palindrome(int number){
int firstNumber = number / 10000;
int secondNumber = (number / 1000) % 10;
int thirdNumber = (number / 100) % 10;
int fourthNumber = (number / 10) % 10;
int fifthNumber = number % 10;

if(firstNumber == fifthNumber && secondNumber == fourthNumber){
return true;
}

return false;
}
public static int factorialOf(int number){
int factorial = 1;
for(int count =1; count <= number;count++){
factorial *= count;

}

return factorial;
}

public static long squareNumber(long number){

long square = number * number;

return square;
}

public static boolean isSquare(int number){
for(int counter = 1; counter < number; counter++){
if(counter * counter == number){

return true;
}


}
return false;
}





}
